app/views/foreman/job_templates/update_packages_by_search_query_-_katello_ansible_default.erb

```diff
@@ -16,14 +16,14 @@ template_inputs:
   required: false
   value_type: plain
 %>
+<% if input('Packages search query').empty? && input('Selected update versions').empty? -%>
+<%= render_template('Package Action - Ansible Default', :state => 'latest', :name => '"*"') %>
+<% else -%>
```

Comment:
I wonder if we need a third case for "only the version field is empty" aka. a user only inputs a package name. Doing this results in a different template for EL vs DEB:

![image](https://github.com/Katello/katello/assets/12595287/a281538a-115b-4e41-80b3-6c898f2330a0)

Both jobs succeed; tested on Foreman 3.9 with input "subscription-manager".

What do you think? @nadjaheitmann

Reply:
Rendering the template with the same inputs should result in updates to the same packages on EL and Debian.

Wouldn't the `name:       []` cause an update to all packages, not just `python3-subscription-manager` ?

Reply:
No, the packages search query is a scoped search and should allow all the same searches as the Packages page at `/packages`.

Reply:
Then I think that the template needs some input validation check. The example above uses `subscription-manager` as input, which is not a valid search query as you said. I think this is out of scope for this PR, though.

Reply:
`subscription-manager` is indeed a valid scoped search input! The code looks fine to me; I was just questioning the screenshots since one has `name:       []` and the other one shows the package names.

Reply:
What I meant is that we have a package name for the `Update Package - Katello Ansible Default` template: 
https://github.com/Katello/katello/blob/master/app/views/foreman/job_templates/update_package_-_katello_ansible_default.erb#L8

And we have `Packages search query` as input for the `Update packages by search query - Katello Ansible Default` template: https://github.com/Katello/katello/blob/master/app/views/foreman/job_templates/update_packages_by_search_query_-_katello_ansible_default.erb#L9

What I am wondering is that if you just enter a package name for the `Packages search query` as input, is this really valid? What I see from the auto-generated inputs is something like `id ^ (1, 2, 3)` when I trigger package update in the new Host details page.

Reply:
Yes, this is a valid search. Searches without key/value pairs are totally legit and they're even the first example in [the scoped_search usage doc](https://github.com/wvanbergen/scoped_search?tab=readme-ov-file#usage-). In the UI, this search would return one or more packages, and then the user would select them with the checkboxes. When clicked, the checkboxes [add the ID](https://github.com/Katello/katello/blob/95a55e0da6b5355170a23bed51d6f5065c05cb7f/webpack/components/extensions/HostDetails/Tabs/PackagesTab/PackagesTab.js#L573) to the selection set; that's why you end up with the `id ^ (1, 2, 3)` in the end.

Reply:
What I'm still not sure about is why this produces different results on Debian. As I said, the existing code here looks like it should work fine.. :thinking:

Reply:
I think I am discovering what's going on. `yum_names_for_job_template` checks `Rpm.installable_for_host` [here](https://github.com/Katello/katello/blob/a1c864acdde85deaff5b22e131086cff5e3ffb4a/app/models/katello/concerns/host_managed_extensions.rb#L526) (which actually returns a list of installed, upgradable packages) and builds a hash of upgradable packages with versions/arches. And `deb_names_for_job_template` just bases it on the name alone, without the version complexities. I suppose for some reason `subscription-manager` wasn't in Katello's list of installed debs for that host, and that's why it returned an empty list.

If you want to add the ability later to select which deb version to upgrade to, a similar change would be needed for `deb_names_for_job_template`. But I'm okay leaving this as-is in this PR.

Reply:
> If you want to add the ability later to select which deb version to upgrade to, a similar change would be needed for deb_names_for_job_template

Eventually :) 

@jeremylenz Thanks for the detailed explanation and the investigation!